### PR TITLE
More tweaks to scripts

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

File: integration/common.sh [hunk 1]
```diff
@@ -23,23 +23,46 @@ function runTest() {
 function waitForServerStartup() {
```
Comment (codingllama) [inline integration/common.sh]: FYI, scripts without the "function" keyword are more portable. Eg, "foo() {}" is better than "function foo() {}"

File: integration/common.sh [hunk 2]
```diff
@@ -23,23 +23,46 @@ function runTest() {
...
   # The server will 404 the request as there's no handler for it. This error doesn't matter
   # as the test will fail if the server is really not up.
-  PORT=$1
+  local PORT=$1
```
Comment (codingllama) [inline integration/common.sh]: nit: lowercase "port"?

(+1 for local)

https://google.github.io/styleguide/shell.xml#Variable_Names

File: integration/common.sh [hunk 3]
```diff
@@ -23,23 +23,46 @@ function runTest() {
...
   wget -q --spider --retry-connrefused --waitretry=1 -t ${STARTUP_WAIT_SECONDS} localhost:${PORT}
   # Wait a bit more to give it a chance to become actually available e.g. if Travis is slow
   sleep 2
 }
 
+function killPid() {
+  local pid=$1
+  set +e
+  while kill -INT ${pid} > /dev/null
```
Comment (codingllama) [inline integration/common.sh]: nit: "while ...; do" on the same line

(I'm not entirely sure if we want to follow the guides, so let me know.)

https://google.github.io/styleguide/shell.xml#Loops

File: integration/common.sh [hunk 4]
```diff
@@ -23,23 +23,46 @@ function runTest() {
...
+  do
+    sleep 1
+    ((count++))
```
Comment (codingllama) [inline integration/common.sh]: nit: Define count outside the loop, so it gets properly scoped?

local count=0
while ...; do
  # ...
done

File: integration/common.sh [hunk 5]
```diff
@@ -23,23 +23,46 @@ function runTest() {
...
 function onExit() {
-  if [[ "${TO_KILL}" ]]; then
-    echo "Killing ${TO_KILL} on exit"
-    kill -INT ${TO_KILL}
-  fi
-  if [[ "${TO_DELETE}" ]]; then
-    echo "Deleting ${TO_DELETE} on exit"
-    rm ${TO_DELETE}
-  fi
+  for pid in "${TO_KILL[@]}"
```
Comment (codingllama) [inline integration/common.sh]: nit: Ditto for the loops here

for ...; do

Same for other files.

File: integration/ct_integration_test.sh [hunk 6]
```diff
@@ -3,6 +3,11 @@ set -e
 INTEGRATION_DIR="$( cd "$( dirname "$0" )" && pwd )"
 . "${INTEGRATION_DIR}"/common.sh
 
+echo "Building code"
```
Comment (codingllama) [inline integration/ct_integration_test.sh]: nit: Single quotes? We don't need expansion here.

Ditto for others.

Comment (daviddrysdale) [inline integration/ct_integration_test.sh]: I'd prefer to keep the same quote style for all of the `echo`s, whether they do or do not happen to mention a variable.

File: integration/common.sh (hunk 5 above)
Comment (daviddrysdale) [inline integration/common.sh]: Done.

File: integration/common.sh (hunk 4 above)
Comment (daviddrysdale) [inline integration/common.sh]: Done

File: integration/common.sh (hunk 3 above)
Comment (daviddrysdale) [inline integration/common.sh]: Done.

(Might as well follow the guide where possible)

File: integration/common.sh (hunk 2 above)
Comment (daviddrysdale) [inline integration/common.sh]: Done.

File: integration/common.sh (hunk 1 above)
Comment (daviddrysdale) [inline integration/common.sh]: Done throughout

File: integration/common.sh [hunk 7]
```diff
@@ -20,26 +20,51 @@ function runTest() {
...
-  fi
-  if [[ "${TO_DELETE}" ]]; then
-    echo "Deleting ${TO_DELETE} on exit"
-    rm ${TO_DELETE}
-  fi
+declare -a TO_KILL
+declare -a TO_DELETE
+onExit() {
+  local pid=0
+  for pid in "${TO_KILL[@]}"
```
Comment (codingllama) [inline integration/common.sh]: FYI, you're still missing a few "for ...; do" occurrences (not counting existing code).

Comment (daviddrysdale) [inline integration/common.sh]: Apologies; added a pass across integration/*sh

**Comment by gdbelvin** [comment]
I didn't see a change to Travis which also uses the presubmit script.

**Pull request merged**